OpenRA.Mods.Common/Lint/CheckDefaultVisibility.cs

```diff
@@ -33,24 +33,31 @@ void Run(Action<string> emitError, Ruleset rules)
 		{
 			foreach (var actorInfo in rules.Actors)
 			{
-				var count = actorInfo.Value.TraitInfos<IDefaultVisibilityInfo>().Count();
-
-				if (count == 0)
-					emitError($"Actor type `{actorInfo.Key}` does not define a default visibility type!");
-				else if (count > 1)
-					emitError($"Actor type `{actorInfo.Key}` defines multiple default visibility types!");
-				else
+				try
 				{
-					var vis = actorInfo.Value.TraitInfoOrDefault<HiddenUnderShroudInfo>();
-					if (vis != null && vis.Type == VisibilityType.Footprint)
+					var count = actorInfo.Value.TraitInfos<IDefaultVisibilityInfo>().Count();
+
+					if (count == 0)
+						emitError($"Actor type `{actorInfo.Key}` does not define a default visibility type!");
+					else if (count > 1)
+						emitError($"Actor type `{actorInfo.Key}` defines multiple default visibility types!");
+					else
 					{
-						var ios = actorInfo.Value.TraitInfoOrDefault<IOccupySpaceInfo>();
-						if (ios == null)
-							emitError($"Actor type `{actorInfo.Key}` defines VisibilityType.Footprint in `{vis.GetType()}` but has no IOccupySpace traits!");
-						else if (ios.OccupiedCells(actorInfo.Value, CPos.Zero).Count == 0)
-							emitError($"Actor type `{actorInfo.Key}` defines VisibilityType.Footprint in `{vis.GetType()}` but does not have any footprint cells!");
+						var vis = actorInfo.Value.TraitInfoOrDefault<HiddenUnderShroudInfo>();
+						if (vis != null && vis.Type == VisibilityType.Footprint)
+						{
+							var ios = actorInfo.Value.TraitInfoOrDefault<IOccupySpaceInfo>();
+							if (ios == null)
+								emitError($"Actor type `{actorInfo.Key}` defines VisibilityType.Footprint in `{vis.GetType()}` but has no IOccupySpace traits!");
+							else if (ios.OccupiedCells(actorInfo.Value, CPos.Zero).Count == 0)
+								emitError($"Actor type `{actorInfo.Key}` defines VisibilityType.Footprint in `{vis.GetType()}` but does not have any footprint cells!");
+						}
 					}
 				}
+				catch (InvalidOperationException e)
+				{
+					emitError($"{e.Message} (Actor type `{actorInfo.Key}`)\n{e.StackTrace}");
```

Comment:
Good idea, but I suggest avoiding the complete stacktrace. It is hard to read, especially when multiple errors occur.

Reply:
Also the errors above start with `Actor type {actorInfo.Key}` instead of ending with it.

Reply:
Sure, I had the same idea of removing stack traces after making this PR. It's not useful for modders.

And as for where Actor type is, I've seen it in errors be either in the middle or end of the message. But I don't mind putting it at start

Reply:
removed stack trace